packages/commonwealth/client/scripts/views/components/TemplateActionSelector/templateActionSelector.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { CWTextInput } from '../component_kit/cw_text_input';
+import { QueryList } from '../component_kit/cw_query_list';
+import { TemplateSelectorItem } from './templateActionSelectorItem';
+import Thread from 'models/Thread';
+import app from 'state';
+
+type TemplateSelectorProps = {
+  onSelect: (template: any) => void;
+  tempTemplates: Array<Pick<any, 'identifier' | 'title'>>;
+  thread: Thread;
+  isOpen: boolean;
+  contracts: Array<any>;
+};
+
+export const TemplateSelector = ({
+  onSelect,
+  tempTemplates,
+  isOpen,
+  contracts,
+}: TemplateSelectorProps) => {
+  // Add a new state for contracts
+  const [loading, setLoading] = useState(false);
+  const [fetched, setFetched] = useState(false);
```

Comment:
'fetched' is assigned a value but never used.
Probably you dont need this line at all